tests/unit/plugins/modules/test_redfish_storage_volume.py

```diff
@@ -608,3 +742,60 @@ def test_check_mode_validation(self, redfish_connection_mock_for_storage_volume,
+
+    def test_check_raid_type_supported_success_case01(self, mocker, redfish_response_mock, storage_volume_base_uri,
+                                                      redfish_connection_mock_for_storage_volume):
+        param = {"raid_type": "RAID0", "controller_id": "controller_id"}
+        f_module = self.get_module_mock(params=param)
+        redfish_response_mock.success = True
+        redfish_response_mock.json_data = {'StorageControllers': [{'SupportedRAIDTypes': ['RAID0', 'RAID6', 'RAID60']}]}
+        self.module.check_raid_type_supported(f_module,
+                                              redfish_connection_mock_for_storage_volume)
+
+    def test_check_raid_type_supported_success_case02(self, mocker, redfish_response_mock, storage_volume_base_uri,
+                                                      redfish_connection_mock_for_storage_volume):
+        param = {"volume_type": "NonRedundant", "controller_id": "controller_id"}
+        f_module = self.get_module_mock(params=param)
+        redfish_response_mock.success = True
+        redfish_response_mock.json_data = {'StorageControllers': [{'SupportedRAIDTypes': ['RAID0', 'RAID6', 'RAID60']}]}
+        self.module.check_raid_type_supported(f_module,
+                                              redfish_connection_mock_for_storage_volume)
+
+    def test_check_raid_type_supported_success_case03(self, mocker, redfish_response_mock, storage_volume_base_uri,
+                                                      redfish_connection_mock_for_storage_volume):
+        param = {"raid_type": "RAID6", "controller_id": "controller_id"}
+        f_module = self.get_module_mock(params=param)
+        redfish_response_mock.success = True
+        redfish_response_mock.json_data = {'StorageControllers': [{'SupportedRAIDTypes': ['RAID0', 'RAID6', 'RAID60']}]}
+        self.module.check_raid_type_supported(f_module,
+                                              redfish_connection_mock_for_storage_volume)
+
+    def test_check_raid_type_supported_success_case04(self, mocker, redfish_response_mock, storage_volume_base_uri,
+                                                      redfish_connection_mock_for_storage_volume):
+        param = {"raid_type": "RAID60", "controller_id": "controller_id"}
+        f_module = self.get_module_mock(params=param)
+        redfish_response_mock.success = True
+        redfish_response_mock.json_data = {'StorageControllers': [{'SupportedRAIDTypes': ['RAID0', 'RAID6', 'RAID60']}]}
+        self.module.check_raid_type_supported(f_module,
+                                              redfish_connection_mock_for_storage_volume)
+
+    def test_check_raid_type_supported_failure_case(self, mocker, redfish_response_mock, storage_volume_base_uri,
+                                                    redfish_connection_mock_for_storage_volume):
+        param = {"raid_type": "RAID9", "controller_id": "controller_id"}
```

Comment:
There's no raid type called RAID9, I know we are just mocking, but better use RAID1 which is not in list of 787 line. just a suggestion.

Reply:
Purposely it was given RAID9 as that is not supported. also its a failure case